### DISCUSSION TITLE
Is there a way to update the app without loading the whole HoF again?

### DISCUSSION THREAD

**Discussion by Liizzeey** [the-marenga/sf-scrapbook-helper#97] [Q&A] [answered]
So in theory: I attack the first one that is given to me is there now a way that the program realizes that I got the 3 items and uses this knowledge to give me new enemies? Like an update function or do I have to fetch the whole HoF every time after 1 attack what doesn't seem like a good idea to me and probably not like the program was intended 😅

**Comment by the-marenga** [comment]
If I understand your request correctly, then that is how the tool is already working. It keeps track of which items you have and recalculates the best enemies, if anything changes (you attack seomone). If that is somehow not working for you, please open an issue with detailed information & logs about your case

**Reply by Liizzeey** [reply]
Okay, I guess I was just too impatient. Thanks for the answer.

**Reply by Liizzeey** [reply]
Okay one question. Do I have to restart the program to reset the list? Because I have the problem when I attack someone this person doesn't disappear from the list until I restart the program and it reloads the list. Is this behaviour correct?

**Reply by the-marenga** [reply]
No, when you fight an enemy, the list will recalculate in a few milliseconds. Of course this only has an effeect, when you win against an enemy, so if you see a red bar pop up on the left, indicating, that you lost, then the list if not going to change

**Reply by the-marenga** [reply]
If that somehow does not happen, please file an issue and submit your log file

**Reply by Liizzeey** [reply]
> No, when you fight an enemy, the list will recalculate in a few milliseconds. Of course this only has an effeect, when you win against an enemy, so if you see a red bar pop up on the left, indicating, that you lost, then the list if not going to change

Does this also apply if I attack manually? Because that is what I am doing, searching the opponent and attack manually 😅

**Reply by the-marenga** [reply]
~~yes, that should work as well. I actually just tested that yesterday and it worked fine then. If it does not for you, then something might be wrong~~

**Reply by the-marenga** [reply]
> Does this also apply if I attack manually? 

Or wait, do you mean you actually open the browser and attack from there? Why?

**Reply by the-marenga** [reply, marked as answer]
That would in fact not update the list. You can enable the "keep characters logged in" option from the setting to force the characters to login again, or start an attack from within the tool, if you want to update them manually. But I don't see, why you would not just press the attack button from withing the helper tool to begin with. That is easier for you and does not have any synchronisation issues

**Reply by Liizzeey** [reply]
I don't know guess it feels saver that Playa doesn't catch on that I use a 3rd party tool, but guess I have to live with it then.

**Reply by the-marenga** [reply]
don't worry, playa does not care. I have never received any mention, that someone using this tool got banned. Playa has 100 different ways of catching people, that use the mfbot 24/7 and they barely do anything against those, so doing a few attacks once will not get you banned.
